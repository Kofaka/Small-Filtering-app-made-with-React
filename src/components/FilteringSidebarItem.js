import React from 'react';

const FilteringSidebarItem = ({tag, label, clickHandler, defaultChecked}) => (
    <div className="filtering-sidebar__item">
        <input
            className="filtering-sidebar__item__icon"
            type="radio"
            id={tag}
            name="filter"
            value={tag}
            onClick={() => clickHandler()}
            defaultChecked={defaultChecked}
        />

        <label
            className="filtering-sidebar__item__name"
            htmlFor={tag}
        >
            {label}
        </label>
    </div>
);

export default FilteringSidebarItem;

