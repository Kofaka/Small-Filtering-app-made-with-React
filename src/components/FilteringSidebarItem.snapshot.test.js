import React from 'react';
import expect from 'expect';
import renderer from 'react-test-renderer';

import FilteringSidebarItem from './FilteringSidebarItem';

const setup = (propsOverrides) => {
    const props = Object.assign({
        tag: 'tag-name-1',
        label: 'SDK 1',
        defaultChecked: false,
        clickHandler: () => {},
    }, propsOverrides);

    return {
        props
    };
};


describe('Component: FilteringSidebarItem', () => {
    it('renders without crashing', () => {
        const {props} = setup();
        const filteringSidebarItem = renderer
            .create(<FilteringSidebarItem {...props} />)
            .toJSON();
        expect(filteringSidebarItem).toMatchSnapshot();
    });
});

