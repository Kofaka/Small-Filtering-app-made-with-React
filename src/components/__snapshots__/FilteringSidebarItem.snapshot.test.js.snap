// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: FilteringSidebarItem renders without crashing 1`] = `
<div
  className="filtering-sidebar__item"
>
  <input
    className="filtering-sidebar__item__icon"
    defaultChecked={false}
    id="tag-name-1"
    name="filter"
    onClick={[Function]}
    type="radio"
    value="tag-name-1"
  />
  <label
    className="filtering-sidebar__item__name"
    htmlFor="tag-name-1"
  >
    SDK 1
  </label>
</div>
`;
